tests/unit/backend/models/admin_verification.server.model.spec.ts

```diff
@@ -0,0 +1,270 @@
+import { ObjectID } from 'bson'
+import mongoose from 'mongoose'
+
+import getAdminVerificationModel from 'src/app/models/admin_verification.server.model'
+import {
+  IAdminVerification,
+  UpsertOtpParams,
+} from 'src/types/admin_verification'
+
+import dbHandler from '../helpers/jest-db'
+
+const AdminVerification = getAdminVerificationModel(mongoose)
+
+describe('AdminVerification Model', () => {
+  beforeAll(async () => await dbHandler.connect())
+  beforeEach(async () => await dbHandler.clearDatabase())
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  describe('Schema', () => {
+    const DEFAULT_PARAMS: IAdminVerification = {
+      admin: new ObjectID(),
+      expireAt: new Date(),
+      hashedContact: 'mockHashedContact',
+      hashedOtp: 'mockHashedOtp',
+    }
+    it('should create and save successfully', async () => {
+      // Act
+      const actual = await AdminVerification.create(DEFAULT_PARAMS)
+
+      // Assert
+      // All fields should exist
+      // Object Id should be defined when successfully saved to MongoDB.
+      expect(actual._id).toBeDefined()
+      expect(actual.createdAt).toBeInstanceOf(Date)
+      expect(actual.updatedAt).toBeInstanceOf(Date)
+      expect(actual).toEqual(
+        expect.objectContaining({
+          ...DEFAULT_PARAMS,
+          // Add defaults
+          numOtpAttempts: 0,
+          numOtpSent: 0,
+        }),
+      )
+    })
+
+    it('should create and save successfully with defaults overriden', async () => {
+      // Arrange
+      const customParams: IAdminVerification = {
+        ...DEFAULT_PARAMS,
+        numOtpAttempts: 10,
+      }
+
+      // Act
+      const actual = await AdminVerification.create(customParams)
+
+      // Assert
+      // All fields should exist
+      // Object Id should be defined when successfully saved to MongoDB.
+      expect(actual._id).toBeDefined()
+      expect(actual.createdAt).toBeInstanceOf(Date)
+      expect(actual.updatedAt).toBeInstanceOf(Date)
+      expect(actual).toEqual(
+        expect.objectContaining({
```

Comment:
does it need to be a nested expect? can't you use `toMatchObject`

Reply:
Done. Just needed to transform with `actual.toObject()`